### PR TITLE
radicle-cob: ObjectIdentifier

### PR DESCRIPTION
**Pull request by FintanH**

### CODE REVIEW

File: radicle-cob/src/object/collaboration.rs [hunk 1]
```diff
@@ -25,28 +25,80 @@ pub use list::list;
...
 pub use update::{update, Update};
 
+/// The full object identifier for a [`CollaborativeObject`].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ObjectIdentifier {
+    /// The [`TypeName`] for the given [`CollaborativeObject`].
+    pub name: TypeName,
+    /// The [`ObjectId`] for the given [`CollaborativeObject`].
+    pub object: ObjectId,
+}
```
Comment (cloudhead) [inline radicle-cob/src/object/collaboration.rs:35]: Hmm, I feel like ObjectIdentifier stands for ObjectId; ie. they are too close and confusable. Some potential suggestions:

* `TypedObjectId`
* `TypedId`
* `TObjectId`

File: radicle-cob/src/change_graph.rs [hunk 2]
```diff
@@ -15,7 +15,10 @@ use petgraph::{
 };
 
 use crate::{
-    change, object, signatures::Signatures, Change, CollaborativeObject, ObjectId, TypeName,
+    change,
+    object::{self, ObjectIdentifier},
+    signatures::Signatures,
+    Change, CollaborativeObject, ObjectId, TypeName,
```
Comment (slack-coder) [inline radicle-cob/src/change_graph.rs:21]: Not particular to this PR, we should come to consensus on how imports are defined

File: radicle-cob/src/object/collaboration.rs (hunk 1 above)
Comment (slack-coder) [reply radicle-cob/src/object/collaboration.rs:35]: Or ObjectMetadata?  The TypeName is descriptive right?  Or wow does the type name affect the identity of the object?  This domain is new to me.

Comment (cloudhead) [reply radicle-cob/src/object/collaboration.rs:35]: It basically describes the type of object refered to by the object-id, so like `(Issue, 123)`

Comment (FintanH) [reply radicle-cob/src/object/collaboration.rs:35]: What about `CobId`, since it's the combination of the two that identifies a `CollaborativeObject`?

Comment (cloudhead) [reply radicle-cob/src/object/collaboration.rs:35]: That could work, though there isn't as clear a relation betwen ObjectId and CobId :thinking:

Comment (cloudhead) [reply radicle-cob/src/object/collaboration.rs:35]: Interestingly though, I haven't found that I "missed" that type while porting some of the code so far, so I wonder if we should scrap the type idea, and just put the function in some module hierarchy that makes sense?

Comment (slack-coder) [reply radicle-cob/src/object/collaboration.rs:35]: @cloudhead I'm not quite understanding, could you make the idea more concrete?

**Pull request closed without merging**